packages/wagmi/src/index.ts

```diff
@@ -23,11 +27,31 @@ export function bindings(): IBindings {
         }
       }
 
-      const signer = await fetchSigner<RequiredSigner>({ chainId });
+      const walletClient = await getWalletClient({ chainId });
 
-      invariant(signer, 'Cannot get signer, is the wallet connected?');
+      invariant(walletClient, 'Wallet client not found');
 
-      return signer;
+      return signerFromWalletClient({ walletClient, chainId });
     },
   };
 }
+
+export function providerFromPublicClient({
+  publicClient,
+  chainId,
+}: {
+  publicClient: PublicClient;
+  chainId?: number;
+}): providers.Web3Provider {
+  return new providers.Web3Provider(publicClient.transport, chainId);
+}
+
+export async function signerFromWalletClient({
+  walletClient,
+  chainId,
+}: {
+  walletClient: WalletClient;
+  chainId?: number;
+}): Promise<providers.JsonRpcSigner> {
+  return new providers.Web3Provider(walletClient.transport, chainId).getSigner();
+}
```

Comment:
What is the mindset in exporting these 2 functions `signerFromWalletClient` and `providerFromPublicClient`?

Reply:
Yeah, I've noticed...  not sure if it's fully Lens SDK responsibility. The example app uses `WebBundlr` almost for convenience, but by design not tied to any specific upload strategy.

Is it possible to rewrite the uploader so to use wagmi/view directly?